sodium/src/hash.rs

```diff
@@ -0,0 +1,88 @@
+//! This module provides access to libsodium
+
+use super::{check_init, secbuf::SecBuf};
+use crate::error::SodiumResult;
+
+/// Size of return value while converting to sha256
+pub const BYTES256: usize = rust_sodium_sys::crypto_hash_sha256_BYTES as usize;
+
+/// Size of return value while converting to sha512
+pub const BYTES512: usize = rust_sodium_sys::crypto_hash_sha512_BYTES as usize;
+
+/// Compute the sha256 hash of input buffer
+/// ****
+/// @param {SecBuf} input - the data to hash
+///
+/// @param {SecBuf} output - Empty Buffer to be used as output
+pub fn sha256(input: &mut SecBuf, output: &mut SecBuf) -> SodiumResult<()> {
+    check_init();
+    let input_len = input.len() as libc::c_ulonglong;
+    let input = input.read_lock();
+    let mut output = output.write_lock();
+    unsafe {
+        rust_sodium_sys::crypto_hash_sha256(
+            raw_ptr_char!(output),
+            raw_ptr_char_immut!(input),
+            input_len,
+        );
+    }
+    Ok(())
+}
+
+/// Compute the sha512 hash of input buffer
+/// ****
+/// @param {Buffer} input - the data to hash
+///
+/// @param {SecBuf} output - Empty Buffer to be used as output
+pub fn sha512(input: &mut SecBuf, output: &mut SecBuf) -> SodiumResult<()> {
+    check_init();
+    let input = input.read_lock();
+    let mut output = output.write_lock();
+    let input_len = input.len() as libc::c_ulonglong;
+    unsafe {
+        rust_sodium_sys::crypto_hash_sha256(
```

Comment:
lol, was wondering why there were so many zeroes in the unit test output below : ) this should be `sha512`.